Feat add service, @input, @output

Ildar-Safin opened the pull request:
Выполнил дз по 10 уроку. Сервисы, передача данных и обработка событий через @Input, @Output
tg: @S_ildar_Abd

Ildar-Safin committed:
Feat. Add new changes.
Выполнил домашнее задание по 7 уроку.
Tg @S_ildar_Abd

Ildar-Safin committed:
fix
исправил первое задание, посмотрел созвон и увидел ошибку у себя и решил исправить
телеграмм @S_ildar_Abd

Ildar-Safin committed:
Выполнил ДЗ по 8 уроку.
тг: @S_ildar_Abd

Ildar-Safin committed:
add new function

Ildar-Safin committed:
add new function

pocahontassss commented on src/app/users-list/users-list.component.html:
```diff
@@ -1,12 +1,8 @@
-    <p>{{$any(item).website}}</p>
-    <p>{{$any(item).company.name}}</p>
-    <button class="click" (click)="deleteUser(item.id)">УДАЛИТЬ ЮЗЕРА</button>
+<div style="display: flex; width: 100%; flex-wrap: wrap">
```
Лучше здесь никогда не писать стили. Всегда пишем их в классах в  scss файле

pocahontassss commented on src/app/users-list/user-card/user-card.component.ts:
```diff
@@ -0,0 +1,20 @@
+})
+
+export class UserCardComponent {
+  @Input()
+  user: any;
```
Протипизировать

pocahontassss commented on src/app/todo-list/todos-list.component.ts:
```diff
@@ -0,0 +1,43 @@
+      }
+    )
+  }
+  deleteTodo(id: any) {
+    this.todos = this.todos.filter(
```
Типизировать

pocahontassss commented on src/app/todo-list/todos-list.component.html:
```diff
@@ -0,0 +1,8 @@
+
+КОМПОНЕНТ СПИСКА ЗАДАЧ
+
+<div style="display: flex; width: 100%; flex-wrap: wrap;">
```
Аналогично. Здесь не стилизуем

pocahontassss commented on src/app/todo-list/todo-card/todo-card.component.ts:
```diff
@@ -0,0 +1,22 @@
+})
+
+export class TodoCardComponent {
+  @Input()
+  todo: any;
+
```
Типизация

Ildar-Safin committed:
fix - style and typing

Ildar-Safin commented:
1. Перенес стили  scss файлы
2. Попробовал сделать типизацию, правильно или нет не знаю, просьба дать обратную связь.

anatolyeviich commented on src/app/users-list/user-card/user-card.component.ts:
```diff
@@ -9,10 +29,10 @@ import {Component, EventEmitter, Input, Output} from "@angular/core";
 export class UserCardComponent {
   @Input()
-  user: any;
+  user: User | undefined;
```
User у тебя undefined не будет. Ошибка уйдет, если ты напишешь "user:! User", так ты скажешь ts'y, что данные будут инициализированы позже

anatolyeviich commented on src/app/users-list/user-card/user-card.component.ts:
```diff
@@ -1,5 +1,25 @@
+    bs: string;
+  }
+}
+
```
В идеале - вынести это в отдельный файл нужно. user-interface.ts

anatolyeviich commented on src/app/users-list/user-card/user-card.component.html:
```diff
@@ -3,5 +3,5 @@
   <p>{{$any(user).website}}</p>
   <p>{{$any(user).company.name}}</p>
-  <button class="click" (click)=onDeleteUser(user.id)>УДАЛИТЬ ЮЗЕРА</button>
+  <button class="click" (click)=onDeleteUser(user!.id)>УДАЛИТЬ ЮЗЕРА</button>
```
А вот здесь просто user.id

anatolyeviich commented on src/app/todo-list/todo-card/todo-card.component.ts:
```diff
@@ -11,10 +18,10 @@ import {NgFor} from "@angular/common";
 export class TodoCardComponent {
   @Input()
-  todo: any;
+  todo: Todo | undefined;
```
Аналогично, todo!

anatolyeviich commented on src/app/todo-list/todo-card/todo-card.component.ts:
```diff
@@ -1,6 +1,13 @@
+  id: number,
+  title: string,
+  completed: boolean
+}
```
Вынести

anatolyeviich commented on src/app/todo-list/todo-card/todo-card.component.html:
```diff
@@ -1,11 +1,11 @@
-  <div>{{todo.completed ? 'ДА' : 'НЕТ'}}</div>
-  <button (click)=onDeleteTodo(todo.id)>УДАЛИТЬ ЗАДАЧУ</button>
+  <div>{{todo!.completed ? 'ДА' : 'НЕТ'}}</div>
+  <button (click)=onDeleteTodo(todo!.id)>УДАЛИТЬ ЗАДАЧУ</button>
```
просто todo.id

Ildar-Safin committed:
fix: typing
Интерфейс вынес в отдельный файл и подключил к компонентам
Исправил ошибки которые были указаны
Благодарю!

Ildar-Safin commented:
интерфейс переписал по новой, User  и Todo в один файл записал и потом с этого файла делал импортировал. Пробовал добавить изменения уже в опубликованный коммит, не получилось по всей видимости, значит что-то делал не так.
